connectivity/drivers/wifi/COMPONENT_WHD/minimal-cyhal/TARGET_STM32H7/cyhal_sdio.c

```
@@ -312,91 +369,77 @@ cy_rslt_t cyhal_sdio_send_cmd(const cyhal_sdio_t *obj, cyhal_transfer_t directio
 {
     uint32_t loop_count = 0;
     cy_rslt_t result;
-    uint16_t attempts = 0;
-    uint32_t temp_sta;
+    uint32_t sta_value;
 
     if (response != NULL) {
         *response = 0;
     }
     current_command = 0;
 
-restart:
     SDIO->ICR = (uint32_t) 0xFFFFFFFF;
-    ++attempts;
-
-    /* Check if we've tried too many times */
-    if (attempts >= (uint16_t) BUS_LEVEL_MAX_RETRIES) {
-        /* WWD_SDIO_RETRIES_EXCEEDED */
-        result = -1;
-        goto exit;
-    }
 
     /* Send the command */
     SDIO->ARG = argument;
     SDIO->CMD = (uint32_t)(command | SDIO_RESPONSE_SHORT | SDIO_WAIT_NO | SDIO_CPSM_ENABLE);
     loop_count = (uint32_t) COMMAND_FINISHED_CMD52_TIMEOUT_LOOPS;
     do {
-        temp_sta = SDIO->STA;
+        sta_value = SDIO->STA;
         loop_count--;
-        if (loop_count == 0 /*|| ((response != NULL) && ((temp_sta & SDIO_ERROR_MASK) != 0))*/) {
```

I'm not entirely sure if this was an intentional workaround or not, but the fact that the error check is commented out here is interesting.  I reenabled it, and it seems that some SDIO commands do seem to generate errors repeatably.  I pulled my hair out for a while until I noticed that the original code was just ignoring the errors and continuing on.  I tried that, and it seemed to work perfectly.  I don't really know enough about SDIO to know why this is happening, but it does seem to work, and that's enough for me for now.